docs/design/referral-manager.md

```diff
@@ -0,0 +1,48 @@
+
+# Title: Referral Manager
+
+## Abstract
+
+Right now, referral program only works per satellite meaning a user on us-east-1 satellite can only refer people to join us-ease-1. 
+However, it's crucial for the tardigrade branded satellites to be able to handle cross-satellite referrals. Therefore, we plan to have a program running in GCP along with tardigrade satellites to handle this issue.
+
+## Background
+
+Along with cross-satellite referrals, we are also working on having a sign-up page that will allow users to select which satellite
+they would like to register an account with. Below design is built based on the new sign-up flow.
+
+## Design
+
+####Definition
+*Referral Link* - _A static link that contains a unique referral code_
+
+*Referral Code* - _An uuid generated by referral manager program for each user to be able to participate in our referral program_
+
+#### User Flow
+1. Sign up
+
+    After a user clicks on the create account button:
+    - Satellite sends a request to referral manager for a referral code and store that code along with the user information inside the user table.
```

Comment:
It might be nice to be more specific about where tables are. Is this the correct interpretation?
> After a user clicks on the create account button on Satellite A:
>  - Satellite A asks referral manager B for a brand new referral code. Satellite A stores that referral code in its own database along with the user information in the user table.

Reply:
Yes, you are absolutely right! I should use a letters to represent different players in this flow for sure